explainers/prioritized-task-scheduling.md

```diff
@@ -0,0 +1,883 @@
+    scheduler.postTask(processInput, {priority: 'user-blocking'});
+  });
+}
+```
+
+<hr/>
+
+Similar to how `fetch()` and other async APIs can be controlled with an `AbortController`, tasks can
+be [controlled](#tasksignal-and-taskcontroller) with a `TaskController` by passing its `TaskSignal`
+to `scheduler.postTask()`. Like `AbortSignal`, `TaskSignal` can be used to abort tasks (a
+`TaskSignal` _is_ an `AbortSignal`). Additionally, `TaskSignal` has a `priority` property, which can
+be changed with the associated controller. For more details, see the section on [`TaskSignal` and
+`TaskController`](#tasksignal-and-taskcontroller).
+
+**Example: Controlling tasks with a `TaskSignal`.**
+```js
+function task() {
+  // ... do a bunch of work...
+}
+
+const controller = new TaskController();
+const signal = controller.signal;
+
+scheduler.postTask(task, {signal});
+
+// ... later ...
+
+// Change the priority, e.g. if the viewport content changed.
+controller.setPriority('background');
+
+// Abort all pending tasks associated with the signal.
+// Note: this causes pending promises to be rejected.
+controller.abort();
+```
+
+<hr/>
+
+If signal and priority are both provided, the task will have a fixed priority and use the signal for
+abort. Note that this is equivalent to passing
+[`TaskSignal.any([signal], priority)`](#tasksignalany-specialization).
+
+**Example: Scheduling a task with signal and priority.**
+
+```js
+function task(signal) {
+  // Use the input signal for aborting, but with fixed 'background' priority.
+  scheduler.postTask(otherTask, {signal, priority: 'background'});
+
+  // The above is equivalent to and shorthand for:
+  const newSignal = TaskSignal.any([signal], {priority: 'background'});
+  scheduler.postTask(otherTask, {signal: newSignal});
+}
+```
+
+**Note**: the `signal` option for `scheduler.postTask()` and other `scheduler` APIs is specified as
+an `AbortSignal`, which means either a plain `AbortSignal` or a `TaskSignal` can be used, since
+`TaskSignal` inherits from `AbortSignal`. This allows these APIs to work with existing code that
+uses `AbortSignal`. It also means a `TaskSignal` can be passed as the `signal` option to other APIs
+that take an `AbortSignal`.
+
+```js
+function doSomethingWithAbortSignal(signal) {
+  // This will work if signal is either an AbortSignal or TaskSignal.
+  scheduler.postTask(task, {signal});
+}
+
+// ... somewhere else ...
+
+const controller = new AbortController();
+const signal = controller.signal;
+fetch(someUrl, {signal});
+doSomethingWithAbortSignal(signal);
+```
+
+### `scheduler.yield()`
+
+(See also the [separate `scheduler.yield()` explainer](./yield-and-continuation.md)).
+
+`scheduler.yield()` can be used in any context to yield to the event loop by awaiting the promise it
+returns. The [task continuation](#definitions-and-concepts) &mdash; the code that runs as a
+microtask when the returned promise is resolved &mdash; runs in a new browser task and gives the
```

Comment:
Out of curiosity, is the code in the continuation run during the microtask checkpoint of the new task, or is it run as part of its regular steps?

Reply:
Thanks for clarifying!